### PR TITLE
Add Optimization Online translator

### PR DESCRIPTION
**Pull request by quachpas**
New translator for https://optimization-online.org/. 

From their website (https://optimization-online.org/about/):
> Optimization Online is a repository of Eprints about optimization and related topics. It facilitates quick dissemination of new research of interest to the optimization community. This service is free.
> Optimization Online submissions are electronically handled by a team of volunteer coordinators.

Unfortunately, I have not managed to parse the abstract consistently as the structure is not fixed.

Thanks in advance!

### CODE REVIEW

File: Optimization Online.js
Comment (adam3smith) [inline Optimization Online.js:160]: Let's save the short URL, which will be much more convenient for citing

File: Optimization Online.js
Comment (adam3smith) [inline Optimization Online.js:85]: just use Zotero's built-in  `text(doc, selector)` instead of `querySelector` and `innerText`

File: Optimization Online.js
Comment (adam3smith) [inline Optimization Online.js:95]: ditto on using `text()`

**Review by adam3smith — CHANGES_REQUESTED** [review]
Looks good in general. A couple of minor changes requested, I'll run the auto-linter once you've made thus

File: Optimization Online.js
Comment (quachpas) [reply Optimization Online.js:93]: When I use `text()` instead, it duplicates mathjax in the abstract.
I left it as it is for now, would you know how to avoid that?
See https://optimization-online.org/?p=23021 for an example

**Comment by quachpas** [comment]
Thank you for the review, I have integrated most changes but need some further discussion on the abstract parsing. 
On another note, do I add you to the credits now or will you do it after the review?

File: Optimization Online.js
Comment (quachpas) [reply Optimization Online.js]: Thank you for the snippet @zoe-translates! I ended up using `innerText()` instead, which was simpler, thank you @AbeJellinek 
I also modified it for the title and added a test case.

**Comment by quachpas** [comment]
Thank you for the help! I believe all brought-up issues were resolved in one way or another.
Would it be ok to run the auto-linter @adam3smith?

File: Optimization Online.js
Comment (adam3smith) [inline Optimization Online.js:40]: @AbeJellinek @dstillman  -- at one point do we decide that it's OK if translators don't work for versions anymore? Preprint was introduced in 6.0, almost 18 month ago, I feel like it's fine to just return this as a type now (as is done here) rather than working with the alternates for pre 6.0 versions, thoughts? 
Since this will come up repeatedly with new item types, it'd be good to have an approximate policy/guideline for this.

**Comment by adam3smith** [comment]
One general question for core devs (See e.g. [OSF Preprints](https://github.com/zotero/translators/blob/master/OSF%20Preprints.js#L39) for what we did right after introducing the `preprint` item type), once we have agreement on this, this lgtm -- thanks!

File: Optimization Online.js
Comment (dstillman) [reply Optimization Online.js:40]: Yes, totally fine to break 5.0 compatibility at this point.

**Comment by adam3smith** [comment]
Great, thanks!

**Pull request merged**
